app/models/spree/payment_decorator.rb

```diff
@@ -0,0 +1,29 @@
+Spree::Payment.class_eval do
+
+  after_save :send_risk_notification, if: :is_risky?
+
+  def is_risky?
+    is_cvv_risky? || is_avs_risky? || state == "failed"
```

Comment:
what if the credit card gets rejected by payment gateway? it'd change to 'failed' sending the notification, doesn't it?